packages/rocketchat-livechat/server/hooks/saveAnalyticsData.js

```diff
@@ -0,0 +1,48 @@
+RocketChat.callbacks.add('afterSaveMessage', function(message, room) {
+	// skips this callback if the message was edited
+	if (!message || message.editedAt) {
+		return message;
+	}
+
+	// check if room is livechat
+	if (room.t !== 'l') {
+		return message;
+	}
+
+	RocketChat.models.Rooms.setLastMessageTimeById(room, message);
```

Comment:
I'm not sure about updating the `room` here and below again .. Why not just update once?
I see you're checking `message.token`, but you're doing this in` setLastMessageTimeById` too, so why not just update the analytics data once?